.github/workflows/prof_controller_docker_publish.yaml

```diff
@@ -1,51 +1,49 @@
 name: Build & Publish Profile Controller Docker image
 on:
-  push:
-    branches:
-      - master
-      - v*-branch
-    paths:
-      - components/profile-controller/**
   pull_request:
     branches:
       - master
       - v*-branch
     paths:
       - components/profile-controller/**
+    types:
+      - closed
 
 env:
   DOCKER_USER: kubeflownotebookswg
   IMG: kubeflownotebookswg/profile-controller
   ARCH: linux/ppc64le,linux/amd64
 
 jobs:
-  push_to_registry:
-    name: Build & Push Docker image to Docker Hub
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+  if_merged:
+    if: github.event.pull_request.merged == true
+    push_to_registry:
+      name: Build & Push Docker image to Docker Hub
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
 
-      - name: Login to DockerHub
-        if: github.event_name == 'push'
-        uses: docker/login-action@v2
-        with:
-          username: kubeflownotebookswg
-          password: ${{ secrets.KUBEFLOWNOTEBOOKSWG_DOCKER_TOKEN }}
+        - name: Login to DockerHub
+          uses: docker/login-action@v2
+          with:
+            username: kubeflownotebookswg
+            password: ${{ secrets.KUBEFLOWNOTEBOOKSWG_DOCKER_TOKEN }}
 
-      - name: Setup QEMU
-        uses: docker/setup-qemu-action@v2
+        - name: Setup QEMU
+          uses: docker/setup-qemu-action@v2
 
-      - name: Setup Docker Buildx
-        uses: docker/setup-buildx-action@v2
+        - name: Setup Docker Buildx
+          uses: docker/setup-buildx-action@v2
 
-      - name: Build multi-arch docker image
-        run: |
-          cd components/profile-controller
-          make docker-build-multi-arch
+        - name: Build multi-arch docker image
+          run: |
+            cd components/profile-controller
+            export IMG=kubeflownotebookswg/profile-controller
+            make docker-build-multi-arch
 
-      - name: Build  and push multi-arch docker image
-        if: github.event_name == 'push'
-        run: |
-          cd components/profile-controller
-          make docker-build-push-multi-arch
+        - name: Build  and push multi-arch docker image
+          run: |
+            cd components/profile-controller
+            export IMG=kubeflownotebookswg/profile-controller
```

Comment:
Same here, regarding not re-declaring the IMG

Reply:
fixed